File: LicenseManagementApi/Models/Entities/License.cs
namespace LicenseManagementApi.Models.Entities;

public class License
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid ProductId { get; set; }
    public Guid RsaKeyId { get; set; }
    public string LicenseKey { get; set; } = string.Empty;
    public string LicenseKeyHash { get; set; } = string.Empty;
    public string SignedPayload { get; set; } = string.Empty;
    public string LicenseType { get; set; } = string.Empty;
    public DateTime? ExpirationDate { get; set; }
    public int MaxActivations { get; set; }
    public int CurrentActivations { get; set; }
    public LicenseStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Customer Customer { get; set; } = null!;
    public Product Product { get; set; } = null!;
    public RsaKey RsaKey { get; set; } = null!;
    
    // Many-to-many relationship with SKUs
    public ICollection<LicenseSku> LicenseSkus { get; set; } = new List<LicenseSku>();
}

File: vue-ui-frontend/src/types/license.ts
export interface License {
  id: string
  customerId: string
  customerName: string
  productId: string
  productName: string
  skuId: string
  skuName: string
  rsaKeyId: string
  rsaKeyName: string
  licenseType: string
  status: string
  expirationDate: string | null
  maxActivations: number
  currentActivations: number
  licenseKey: string
  createdAt: string
  updatedAt: string
}

export interface CreateLicenseDto {
  customerId: string
  productId: string
  skuId: string
  rsaKeyId: string
  licenseType: string
  expirationDate: string | null
  maxActivations: number
}

export interface UpdateLicenseDto {
  customerId: string
  productId: string
  skuId: string
  rsaKeyId: string
  licenseType: string
  expirationDate: string | null
  maxActivations: number
}
